artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/AddressControlImpl.java

```diff
@@ -373,6 +372,86 @@ public String sendMessage(final Map<String, String> headers,
       return MBeanInfoHelper.getMBeanAttributesInfo(AddressControl.class);
    }
 
+   @Override
+   public void pause() throws Exception {
+      pause(false);
+   }
+
+   @Override
+   public void pause(boolean persist) throws Exception {
+      if (AuditLogger.isEnabled()) {
+         AuditLogger.pause(addressInfo);
+      }
+      checkStarted();
+
+      clearIO();
+      try {
+         if (!addressInfo.isPaused()) {
+            addressInfo.pause();
+            if (persist) {
+               server.getStorageManager().storeAddressStatus(addressInfo.getId(), AddressStatus.PAUSED);
+            }
+            Bindings bindings = server.getPostOffice().lookupBindingsForAddress(addressInfo.getName());
+            if (bindings != null) {
+               for (Binding binding : bindings.getBindings()) {
+                  if (binding instanceof QueueBinding) {
+                     QueueControl coreQueueControl = (QueueControl) managementService.getResource(ResourceNames.QUEUE + binding.getUniqueName().toString());
+                     if (coreQueueControl != null) {
+                        coreQueueControl.pause();
+                     }
+                  }
+               }
+            }
+         }
+      } finally {
+         blockOnIO();
+      }
+   }
+
+   @Override
+   public void resume() throws Exception {
+      resume(false);
+   }
+
+   @Override
+   public void resume(boolean persist) throws Exception {
+      if (AuditLogger.isEnabled()) {
+         AuditLogger.resume(addressInfo);
+      }
+      checkStarted();
+
+      clearIO();
+      try {
+         if (addressInfo.isPaused()) {
+            addressInfo.resume();
+            if (persist) {
+               server.getStorageManager().storeAddressStatus(addressInfo.getId(), AddressStatus.RUNNING);
```

Comment:
This should be pushed into `AddressInfo` as described above. Also, resuming the address should simply delete the existing PAUSED record (if one exists) otherwise the bindings journal will grow unbounded with these records.

Reply:
That means that the storageManager will be in AddressInfo  right ?

Reply:
Yes. That appears to be the most straight-forward solution.

Reply:
the storageManager and the postOffice are not always available where AddressInfo are created. Hum are we sure about this storage feature ?